### ISSUE TITLE
Enhancement: [no-floating-promises] enable "add await" suggestion if `ignoreVoid` is true

### ISSUE THREAD

**Issue by FloEdelmann** [typescript-eslint/typescript-eslint#9685]
### Before You File a Proposal Please Confirm You Have Done The Following...

- [X] I have [searched for related issues](https://github.com/typescript-eslint/typescript-eslint/issues?q=is%3Aissue+label%3A%22enhancement%3A+plugin+rule+option%22) and found none that match my proposal.
- [X] I have searched the [current rule list](https://typescript-eslint.io/rules/#rules) and found no rules that match my proposal.
- [X] I have [read the FAQ](https://typescript-eslint.io/linting/troubleshooting) and my problem is not listed.

### My proposal is suitable for this project

- [X] I believe my proposal would be useful to the broader TypeScript community (meaning it is not a niche proposal).

### Link to the rule's documentation

https://typescript-eslint.io/rules/no-floating-promises/

### Description

When a Promise-returning function is called without `await`ing or `void`ing it, the `no-floating-promises` rule correctly reports it and offers a suggestion to fix the call.

However, that suggestion depends on the `ignoreVoid` rule option:

* if it is `false`, only adding `await` is suggested
* if it is `true`, only adding `void` is suggested

The former case is fine because this is the only way to fix the error.

In the latter case, the error could be fixed by either adding `await` or `void`, and it depends on the context which one is correct.

Therefore, I suggest to additionally enable the "add `await` operator" suggestion if `ignoreVoid` is `true`.

### Fail

```TypeScript
declare function foo(arg: number): Promise<void>

async function bar() {
  // if `ignoreVoid` is `false`:
  foo(2) // suggests "Add await operator."

  // if `ignoreVoid` is `true`:
  foo(2) // suggests "Add void operator to ignore."
}
```

### Pass

```TypeScript
declare function foo(arg: number): Promise<void>

async function bar() {
  // if `ignoreVoid` is `false`:
  foo(2) // suggests "Add await operator."

  // if `ignoreVoid` is `true`:
  foo(2) // suggests Add await operator." and "Add void operator to ignore."
}
```

### Additional Info

See [the above code in the playground](https://typescript-eslint.io/play#ts=5.3.3&fileType=.tsx&code=CYUwxgNghgTiAEAzArgOzAFwJYHtVJxwApYBzALnlWQFsAjEGASkoAUYcasBnEAHgBuOLMAB8AKHFRuAT3RI0mXPjqwiTeAG9x8AsQBMTcQF9xQA&eslintrc=N4KABGBEBOCuA2BTAzpAXGUEKQAIBcBPABxQGNoBLY-AWhXkoDt8B6Jge1oDN4OBDfMwDmtYtA4BbSshTowAbXDYciaBOiQANGGUrgUSsM7REANQ6UAJvPxxEYAL56wAXWXPHQA&tsconfig=N4KABGBEDGD2C2AHAlgGwKYCcDyiAuysAdgM6QBcYoEEkJemy0eAcgK6qoDCAFutAGsylBm3TgwAXxCSgA&tokens=false) and toggle `ignoreVoid` in the "eslintrc" tab.
